packages/es-components/src/components/base/icons/Icon.js

```diff
@@ -2,48 +2,36 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { regular } from './icon-definitions';
-
 const StyledIcon = styled.i`
   display: inline-block;
-  font-family: bds-func-icons !important;
-  font-size: ${props => props.fontSize};
-  font-style: normal;
-  font-weight: inherit;
-  font-variant: normal;
-  line-height: 1;
-  speak: none;
+  font-size: ${props => props.size};
   text-decoration: none;
-  text-transform: none;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
   vertical-align: text-bottom;
-
-  &:before {
-    content: "\\e${props => props.content}";
-  }
 `;
 
-function Icon({ name, size, ...other }) {
-  const iconName = `im-icon-${name}`;
-
-  const styledIconProps = {
-    content: regular[iconName],
-    fontSize: size !== undefined ? `${size}px` : 'inherit'
-  };
-
-  return <StyledIcon aria-hidden {...styledIconProps} {...other} />;
+function Icon({ name, size, className, ...other }) {
+  return (
+    <StyledIcon
+      className={`bds-icon bds-${name}${className ? ` ${className}` : ''}`}
+      size={size ? `${size}px` : 'inherit'}
```

Comment:
I wonder if this should be a little smarter. Maybe we could do something like:
```js
/^\d$/.test(size) ? `${size}px` : (size || 'inherit')
```

This would allow for stuff like:
```jsx
<Icon name="hello" size="1.4em" />
```

Reply:
`size` is a `number` though, not a string. Would it be a breaking change if we changed the type?
Interestingly, `FullColorIcon` also has `size` as a `number`, but doesn't infer a default unit at all, so it kind of requires one...

Reply:
I suppose adding that logic in there wouldn't necessitate changing the type, it would be more of a hidden safety feature.

Reply:
Another interesting thing I found... `FullColorIcon` seems to work fine without specifying a unit because it's using the `css` prop from styled-components. That seems to be adding `px` as a default if it isn't provided, whereas a full styled component does not. :\

Reply:
I don't think changing the type would be a breaking change if we made it `PropTypes.oneOfType([PropTypes.string, PropTypes.number])`. It's just adding the capability of including other measurement units.

The `css` prop works because it's kind of an expected React pattern to convert a number size on a style object into `px`. The same thing is true with [`style` props on DOM nodes in React](https://reactjs.org/docs/dom-elements.html#style).

Reply:
Very cool, I learned something new today!